### PR TITLE
Update build_setup.rst

### LINKED ISSUE #5249: Update compiling CCF from source instructions
I think https://microsoft.github.io/CCF/main/contribute/build_setup.html needs updating to specific that ``--extra-vars "clang_version=15"`` should be added to ``./run.sh ccf-dev.yml`` for compiling ccf 4.0 in virtual

### PR DESCRIPTION
**Pull request by achamayou**
Closes #5249

### CODE REVIEW

File: doc/contribute/build_setup.rst [hunk 1]
```diff
@@ -11,7 +11,12 @@ Then, to quickly set up the dependencies necessary to build CCF itself and CCF a
 .. code-block:: bash
 
     $ cd <ccf_path>/getting_started/setup_vm
-    $ ./run.sh ccf-dev.yml
+    # For SGX machines
+    $ ./run.sh ccf-dev.yml --extra-vars "platform=sgx"
+    # For SNP machines
+    $ ./run.sh ccf-dev.yml --extra-vars "platform=snp" "clang_version=15"
```
Comment (achamayou) [inline doc/contribute/build_setup.rst:17]:
```suggestion
    $ ./run.sh ccf-dev.yml --extra-vars "platform=snp clang_version=15"
```

File: doc/contribute/build_setup.rst [hunk 2]
```diff
@@ -11,7 +11,12 @@ Then, to quickly set up the dependencies necessary to build CCF itself and CCF a
...
+    # Generic debugging build, _without_ enclave protection
+    $ ./run.sh ccf-dev.yml --extra-vars "platform=virtual" "clang_version=15"
```
Comment (achamayou) [inline doc/contribute/build_setup.rst:19]:
```suggestion
    $ ./run.sh ccf-dev.yml --extra-vars "platform=virtual clang_version=15"
```

File: doc/contribute/build_setup.rst [hunk 3]
```diff
@@ -11,7 +11,12 @@ Then, to quickly set up the dependencies necessary to build CCF itself and CCF a
 .. code-block:: bash
 
     $ cd <ccf_path>/getting_started/setup_vm
-    $ ./run.sh ccf-dev.yml
+    # For SGX machines
```
Comment (eddyashton) [inline doc/contribute/build_setup.rst:14]: Can't apply a suggestion on the whole block, but can we use the per-platform `.. tab:: SGX` pattern that we have in `install_bin.rst` for this?

**Review by eddyashton — APPROVED** [review]

File: doc/contribute/build_setup.rst (hunk 3 above)
Comment (achamayou) [reply doc/contribute/build_setup.rst:14]: You can if you click on the +/- line and drag down in fact.

Comment (eddyashton) [reply doc/contribute/build_setup.rst:14]: But it needs to start from line 11, which is outside the changes so can't be suggested.

Comment (achamayou) [reply doc/contribute/build_setup.rst:14]: Yeah, so it seems.

**Pull request merged**